### PR TITLE
chore: Convert GraphQL meeting types to use codegen

### PR DESCRIPTION
**Pull request by Dschoordsch**
# Description

Relates to #9291 
To simplify adding recurring retros and avoid working on legacy types, convert the GraphQL meeting types to codegen. The old types were left in place empty to avoid having to convert all referencing types as well.

## Demo

[Nothing to see here](https://www.youtube.com/watch?v=NuAKnbIr6TE)

## Testing scenarios

- Smoke test all meeting types

## Final checklist

- [ ] I checked the [code review guidelines](../docs/codeReview.md)
- [ ] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [ ] I have performed a self-review of my code, the same way I'd do it for any other team member
- [ ] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [ ] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [ ] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [ ] PR title is human readable and could be used in changelog

### CODE REVIEW

File: packages/server/graphql/public/mutations/startCheckIn.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import publish from '../../../utils/publish'
+import standardError from '../../../utils/standardError'
+import createGcalEvent from '../../mutations/helpers/createGcalEvent'
+import createNewMeetingPhases from '../../mutations/helpers/createNewMeetingPhases'
+import isStartMeetingLocked from '../../mutations/helpers/isStartMeetingLocked'
+import maybeCreateOneOnOneTeam from '../../mutations/helpers/maybeCreateOneOnOneTeam'
+import {IntegrationNotifier} from '../../mutations/helpers/notifications/IntegrationNotifier'
+import {MutationResolvers} from '../resolverTypes'
+
+const startCheckIn: MutationResolvers['startCheckIn'] = async (
```
Comment (Dschoordsch) [inline packages/server/graphql/public/mutations/startCheckIn.ts:19]: Nothing was changed here

File: packages/server/graphql/public/mutations/startRetrospective.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import {getUserId, isTeamMember} from '../../../utils/authorization'
+import publish from '../../../utils/publish'
+import standardError from '../../../utils/standardError'
+import {MutationResolvers} from '../resolverTypes'
+import createGcalEvent from '../../mutations/helpers/createGcalEvent'
+import createNewMeetingPhases from '../../mutations/helpers/createNewMeetingPhases'
+import isStartMeetingLocked from '../../mutations/helpers/isStartMeetingLocked'
+import {IntegrationNotifier} from '../../mutations/helpers/notifications/IntegrationNotifier'
+
+const startRetrospective: MutationResolvers['startRetrospective'] = async (
```
Comment (Dschoordsch) [inline packages/server/graphql/public/mutations/startRetrospective.ts:20]: No change here

File: packages/server/graphql/public/typeDefs/NewMeeting.graphql [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+  """
+  The auto-incrementing meeting number for the team
+  """
+  meetingNumber: Int!
+
+  """
+  The id of the meeting series this meeting belongs to
+  """
+  meetingSeriesId: ID
```
Comment (Dschoordsch) [inline packages/server/graphql/public/typeDefs/NewMeeting.graphql:63]: A few fields were defined in code but not the `_legacy.graphql`. I've added these here: `locked`, `meetingSeriesId`, `summary` and `scheduledEndTime`.

File: packages/server/graphql/public/types/NewMeeting.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      ...phase,
+      meetingId,
+      teamId
+    }))
+
+    if (locked) {
+      // make all stages non-navigable so even if the user removes the overlay they cannot see all meeting data
+      return resolvedPhases.map((phase: any) => ({
+        ...phase,
+        stages: phase.stages.map((stage: any) => ({
```
Comment (nickoferrall) [inline packages/server/graphql/public/types/NewMeeting.ts:50]: +1 this isn't new code, but the three `any`s could be removed here as they're inferred

**Review by nickoferrall — APPROVED** [review]
Smoke tested and it's all looking good

**Pull request merged**